packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -95,6 +100,20 @@ export default util.createRule<Options, MessageId>({
             context.report({
               node,
               messageId: 'floating',
+              suggest: [
+                {
+                  messageId: 'floatingFixAwait',
+                  fix(fixer): TSESLint.RuleFix {
+                    let code = sourceCode.getText(node);
+                    code = `await ${
+                      code.startsWith('void')
+                        ? code.replace('void', '').trimStart()
+                        : code
+                    }`;
+                    return fixer.replaceText(node, code);
```

Comment:
We generally try to avoid replacing text that isn't actually changed. Anything after a `void` operator doesn't need to be replaced.

I'd suggest using `fixer.replaceTextRange`, so you can target just the `void` operator if it exists. You can then use `replaceTextRange` or even `insertTextBefore` if there isn't a `void` operator.

Reply:
@JoshuaKGoldberg 
Thank you for your kind comments!
I have fixed the method of judging void operator and replace or add await operator.

dcfa4bb05240409f7ceb21b3e4bc05ed1d9c9cbc

Reply:
Love it!